tests/test_core.py

```diff
@@ -290,6 +290,45 @@ def __test(infile):
         yield (__test, infile)
 
 
+def test_istft_reconstruction():
+    from scipy.signal import bartlett, hann, hamming, blackman, blackmanharris
+
+    def __test(x, n_fft, hop_length, window):
+        S = librosa.core.stft(
+            x, n_fft=n_fft, hop_length=hop_length, window=window)
+        x_reconstructed = librosa.core.istft(
+            S, hop_length=hop_length, window=window)
+
+        L = min(len(x), len(x_reconstructed))
+        x = np.resize(x, L)
+        x_reconstructed = np.resize(x_reconstructed, L)
+
+        # NaN should not happen
+        assert not np.any(np.isnan(x_reconstructed))
+
+        # should be almost approximately reconstucted
+        assert np.allclose(x, x_reconstructed, atol=1.0e-7)
+
+    test_file = 'data/test1_44100.wav'
+    x, sr = librosa.load(test_file, sr=None, mono=True)
+
+    for window_func in [bartlett, hann, hamming, blackman, blackmanharris]:
+        for n_fft in [512, 1024, 2048, 4096]:
+            win = window_func(n_fft, sym=False)
+            symwin = window_func(n_fft, sym=True)
+            for hop_length_denom in six.moves.range(2, 10):
+                hop_length = n_fft / hop_length_denom
```

Comment:
``n_fft // hop_length_denom`` to ensure integer division.  I think this is the cause of the log bloat.